components/mpas-ocean/src/shared/mpas_ocn_wetting_drying.F

```diff
@@ -356,40 +363,207 @@ subroutine ocn_wetting_drying_wettingVelocity(layerThickEdgeFlux, layerThickness
+             end do
+             layerThickness = layerThickness + dt * divOutFlux
 
-          call ocn_wetting_velocity_factor_on_cell_edges(wettingVelocityFactor, layerThickness, normalVelocity, iCell, k)
+             call ocn_wetting_velocity_factor_on_cell_edges(wettingVelocityFactor, layerThickness, normalVelocity, iCell, k)
+
+           end do
+         end do
+         !$omp end do
+         !$omp end parallel
+      end if
+
+
+      ! This wetting and drying implementation is based on the one above.
+      ! However, it has special considerations for edges with normal velocities
+      ! that have been previously zeroed to limit drying (or are zero initially),
+      ! but have ssh gradients that would cause flow into the current cell,
+      ! i.e., the neighboring cell's ssh is higher.
+      ! There are two cases that require special treatment:
+      !  1) In the case that the neighbor cell is predicted to be wet, flow should
+      !     be allowed into the current cell.
+      !  2) In the case that the neighor cell is predicted to be dry, flow should
+      !     be prevented completely (and not ramped).
+      ! Currently, this only works for single layer configurations with the 'ssh_gradient'
+      ! pressure gradient option
+
+      if (config_use_modify_wetting_drying_implementation .and. &
+          pGradType == pGradTypeSSHgrad) then
+
+        layerThicknessCell => layerThicknessCellWetDry
+        sshCell => sshCellWetDry
+        layerThickness = 0.0_RKIND
+        sshCell = 0.0_RKIND
+
+        wettingVelocityFactor = 0.0_RKIND
+
+        ! Predict new layerThickness based on out-going flux
+        do iCell = 1, nCellsAll
+           do k = minLevelCell(iCell), maxLevelCell(iCell)
+             divOutFlux = 0.0_RKIND
+             layerThickness = min(layerThicknessProvis(k, iCell), layerThicknessCur(k, iCell))
+             do i = 1, nEdgesOnCell(iCell)
+               iEdge = edgesOnCell(i, iCell)
+               if (k <= maxLevelEdgeTop(iEdge) .and. k >= minLevelEdgeBot(iEdge)) then
+                 ! only consider divergence flux leaving the cell
+                 if ( normalVelocity(k, iEdge) * edgeSignOnCell(i, iCell) < 0.0_RKIND ) then
+                   divOutFlux = divOutFlux + &
+                                normalVelocity(k, iEdge) * edgeSignOnCell(i, iCell) * &
+                                layerThickEdgeFlux(k, iEdge) * dvEdge(iEdge) * &
+                                invAreaCell(iCell)
+                  end if
+               end if
+             end do
+             layerThickness = layerThickness + dt * divOutFlux
+           end do
+
+           layerThicknessCell(iCell) = layerThickness
 
         end do
-      end do
-      !$omp end do
-      !$omp end parallel
+
+        call mpas_dmpar_field_halo_exch(domain, 'layerThicknessCellWetDry')
+
+        ! Compute predicted ssh
+        do iCell = 1, nCellsAll
+           if ( config_use_subgrid_wetting_drying  ) then
```

Comment:
I don't think this is necessary. The reason this `config_use_ssh_gradient_wetting_drying` check is needed in the `config_use_modify_wetting_drying_implementation` (now `config_use_ssh_gradient_wetting_drying`) option is to find the subgrid ssh value from the lookup tables. Since ssh doesn't factor into the standard wetting and drying, there isn't a need for a `config_use_ssh_gradient_wetting_drying` check there.

Reply:
Sounds good